### ISSUE TITLE
Fancy printing and doctest printing clash

### ISSUE THREAD

**Issue by thofma** [oscar-system/Oscar.jl#475]
The following behavior was observed by @wdecker and since I could not find an easy solution, I want to record it here. In the REPL we have the following output:
```julia
julia> R, (x, y) = PolynomialRing(QQ, ["x", "y"])
(Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])

julia> A, _ = quo(R, ideal(R, [x^2-y^3, x-y]))
(Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal generated by: x^2 - y^3, x - y, Map from
Multivariate Polynomial Ring in x, y over Rational Field to A defined by a julia-function with inverse
)
```
Notice the `A` in the second output. We do this by cleverly looking up the bindings in `Main`. Unfortunately, the same example run as a doctest returns
```julia
julia> R, (x, y) = PolynomialRing(QQ, ["x", "y"])
(Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])

julia> A, _ = quo(R, ideal(R, [x^2-y^3, x-y]))
(Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal generated by: x^2 - y^3, x - y, Map from
Multivariate Polynomial Ring in x, y over Rational Field to Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal generated by: x^2 - y^3, x - y defined by a julia-function with inverse
)
```
Notice that there is no `A` in the second output. This happens because doctests are evaluated in independent modules (not `Main`) and the fancy printing looks only at bindings in `Main`.

This is bad because this means adding doctests becomes more complicated in case this happens. There are various options:
1. Remove the fancy printing.
2. Add a global option to disable fancy printing. Just so that one can produce the doctests more easily.
3. Lookup the bindings also in the modules defined during the doctests (I think they have a predictable name like `__doctest*`). This might lead to some wrong printing because one could use the identical (because of caching) object with different names.
4. The simple solution I did not think of.

**Comment by wdecker** [comment]
This is also confusing for the user as what he sees in the docu is different from what he sees in practise.

**Comment by ThomasBreuer** [comment]
We can try to convince the `Documenter.jl` developers that this shows a bug.

**Comment by thofma** [comment]
I don't think it is a bug that doctests are run in their own module/scope. Running them at toplevel would make them pretty fragile.

I had to fix it again by hand in #506. This is tedious. Next time I have to do it I will just disable the fancy printing altogether.

**Comment by fingolfin** [comment]
We have now implemented a solution for this via `AbstractAlgebra.set_current_module`
